In [1]:
# Import spacy
import spacy
nlp = spacy.load('en_core_web_md')

In [2]:
import os
#import codecs

navigate for data access

In [3]:
!pwd

/home/zafrin/PycharmProjects/RateMyProfessor/notebooks


In [4]:
!cd ..

In [5]:
os.chdir('/home/zafrin/PycharmProjects/RateMyProfessor')

In [6]:
!pwd

/home/zafrin/PycharmProjects/RateMyProfessor


In [7]:
directory = os.path.join('data', 'intermediate', 'comments')
filepath = os.path.join(directory, 'comment_sents_raw.txt')
test_filepath = os.path.join(directory, 'comment_sents_test.txt')

##### Testing with NLP

In [9]:
with open(filepath, 'r', encoding='utf-8') as f:
    raw_comments = f.read()

In [11]:
tokens = nlp(raw_comments[:100000])

In [76]:
tokens[10]

things

In [16]:
len(tokens)

23117

In [13]:
raw_comments[0]

'T'

In [17]:
len(raw_comments)

2313064911

100,000 characters from Raw comments was processed, 
which included 23,117 words

In [15]:
for token in tokens.sents:
    print(token.lemma_)

the textbook be almost pointless if -PRON- can reasonably piece thing together from the internet and the online studyspace for the book .
-PRON- be never use in class , only for the homework .
this class should really be consider speak intensive-
-PRON- have a 10 minute presentation every other week .
quizes be very hard for non - music major .
very helpful and understand
i have -PRON- for abnormal
and i love -PRON- .
-PRON- be interesting and alwasy be able to provide an example .
the test be not that bad , but only three of -PRON- .
-PRON- have -PRON- now for abnormal
and -PRON- love -PRON- .
-PRON- think -PRON- be really fair and keep the class interesting .
will do well if -PRON- go to class and actually pay attention .
the lecture jump from place to place .
3 test ( multiple choice and essay ) .
nice guy but the test be very ambiguous 
 nice guy but impossible test ... only give 2 with a ton of stuff .. make question on test confusing , come up with question out of nowhere that -P

the quiz be ambiguos and -PRON- be very difficult to find an exact answer to -PRON- !
there be no exact date state for when thing be due , -PRON- have to pay close attention to what session or week
-PRON- be in .
what ever -PRON- do do not take -PRON- ! !
ridiculous class .
online class , although -PRON- do not think this guy have any sort of computer skill .
essay exam question be very obscure and difficult to find enough information to provide an adequate length of an answer .
do not respond to email so do not plan to correspond with -PRON- .
-PRON- need to retire .
but until then make sure -PRON- take someone else
bad class -PRON- have take .
-PRON- seem to have no computer skills.4 discussion board and 2 essay exam test , all ask impossible , unclear , ridiculous question .
no method of grading and in -PRON- opinion
the only reason why the average of the 2nd essay exam be high be so that the majority of the class do not fail .
-PRON- be suppose to back up -PRON- answer wit
-PRON- o

##### Testing Word2Vac model for raw comments subset|

In [18]:
from gensim import corpora

In [45]:
f = open(test_filepath, 'r', encoding='utf-8')
test_comments = [f.readline()[:-1] for line in f]
test_comments = [sent.split() for sent in test_comments]

In [21]:
type(test_comments)

list

In [22]:
len(test_comments)

11799

In [23]:
test_comments[0]

["It's", 'never', 'used', 'in', 'class,', 'only', 'for', 'the', 'homeworks.']

In [24]:
dictionary = corpora.Dictionary(test_comments)

In [25]:
f.close()

In [26]:
print(dictionary)

Dictionary(12146 unique tokens: ["It's", 'class,', 'for', 'homeworks.', 'in']...)


In [27]:
print(dictionary.token2id)

{"It's": 0, 'class,': 1, 'for': 2, 'homeworks.': 3, 'in': 4, 'never': 5, 'only': 6, 'the': 7, 'used': 8, '10': 9, 'a': 10, 'every': 11, 'had': 12, 'minute': 13, 'other': 14, 'presentation': 15, 'we': 16, 'week.': 17, 'Very': 18, 'and': 19, 'helpful': 20, 'understanding': 21, 'able': 22, 'alwasy': 23, 'an': 24, 'example.': 25, "he's": 26, 'interesting': 27, 'is': 28, 'provide': 29, 'to': 30, 'Abnormal': 31, 'I': 32, 'have': 33, 'him': 34, 'now': 35, 'class': 36, 'fair': 37, 'interesting.': 38, 'keeps': 39, 'really': 40, 'think': 41, 'The': 42, 'from': 43, 'jumps': 44, 'lecture': 45, 'place': 46, 'place.': 47, 'Nice': 48, 'ambiguous': 49, 'are': 50, 'but': 51, 'guy': 52, 'tests': 53, 'very': 54, 'all': 55, 'all,': 56, 'distaster': 57, 'Had': 58, 'concerns.': 59, 'of': 60, 'psych': 61, 'ultimate': 62, 'VERY': 63, 'awesome.': 64, 'guess': 65, 'i': 66, 'ok': 67, 'was': 68, 'He': 69, 'Psychologists.': 70, 'as': 71, 'feel': 72, 'if': 73, 'makes': 74, 'real': 75, 'us': 76, 'AS': 77, 'BORING': 

In [28]:
from gensim.models.word2vec import Word2Vec

In [29]:
type(test_comments)

list

In [30]:
test_comments[6]

['The', 'lecture', 'jumps', 'from', 'place', 'to', 'place.']

In [31]:
model = Word2Vec(test_comments)

In [32]:
model.wv['lecture']

array([-0.00153972, -0.21998699, -0.21219549,  0.14702092, -0.38191748,
       -0.26305345,  0.11623773, -0.03393426, -0.08372637,  0.15724996,
       -0.27767625, -0.1393332 , -0.35682437,  0.4289479 , -0.11316595,
       -0.18556698,  0.18617165,  0.07901307, -0.20697285, -0.310505  ,
       -0.25631276,  0.0872377 , -0.020251  ,  0.00569327,  0.39281976,
        0.14474413, -0.30241853,  0.05232759,  0.10079192,  0.43607238,
        0.09237757, -0.31336269, -0.04709129, -0.04089181, -0.1084859 ,
       -0.1417319 ,  0.03430048,  0.03431017, -0.12908734,  0.19173999,
        0.05062667, -0.12715136, -0.0458032 ,  0.17087238, -0.24359043,
       -0.12045649,  0.17793854,  0.3951984 ,  0.0498809 ,  0.29564899,
       -0.22790514, -0.5087778 , -0.38669059,  0.12862571,  0.2510584 ,
        0.20387764,  0.25095549,  0.27758533,  0.13247268,  0.26518065,
        0.08758297, -0.13581039, -0.00362211, -0.0874488 ,  0.03554257,
        0.35387984,  0.06191468, -0.11379325,  0.09764733,  0.15

In [34]:
model.wv.most_similar('lecture', topn =10)

[('some', 0.9998031854629517),
 ('off', 0.9997953176498413),
 ('final', 0.999763548374176),
 ('reading', 0.9997153282165527),
 ('exam', 0.9996781349182129),
 ('points', 0.9996683597564697),
 ('information', 0.999664843082428),
 ('worth', 0.9996457099914551),
 ('each', 0.9996401071548462),
 ('required', 0.9996331930160522)]

##### Testing Word2Vec model by removing stopwords

In [47]:
for i, comment in enumerate(test_comments):
    lst = []
    for word in comment:
        if nlp.vocab[word].is_stop == False:
            lst.append(word)
    test_comments[i] = lst

In [48]:
test_comments[6]

['The', 'lecture', 'jumps', 'from', 'place', 'to', 'place.']

In [49]:
for sent in test_comments:
    print(sent)

["It's", 'never', 'used', 'in', 'class,', 'only', 'for', 'the', 'homeworks.']
['we', 'had', 'a', '10', 'minute', 'presentation', 'every', 'other', 'week.']
['Very', 'helpful', 'and', 'understanding']
["he's", 'interesting', 'and', 'alwasy', 'is', 'able', 'to', 'provide', 'an', 'example.']
['I', 'have', 'him', 'now', 'for', 'Abnormal']
['I', 'think', "he's", 'really', 'fair', 'and', 'keeps', 'the', 'class', 'interesting.']
['The', 'lecture', 'jumps', 'from', 'place', 'to', 'place.']
['Nice', 'guy', 'but', 'the', 'tests', 'are', 'very', 'ambiguous']
['all', 'and', 'all,', 'a', 'distaster']
['Had', 'him', 'for', 'psych', 'of', 'ultimate', 'concerns.']
['VERY', 'interesting.']
["he's", 'awesome.']
['was', 'ok', 'i', 'guess']
['He', 'makes', 'us', 'feel', 'as', 'if', 'we', 'are', 'real', 'Psychologists.']
['BORING', 'AS', 'HELL!']
['Run', 'in', 'the', 'opposite', 'direction!']
['study', 'for', 'tests', 'if', 'you', 'dont', 'kow', 'about', 'eastern', 'cultures,', 'but', 'if', 'you', 'do,', '

['He', 'can', 'be', 'pretty', 'funny', 'and', 'the', 'tests', "aren't", 'too', 'hard', '-', 'just', 'read', 'the', 'points', 'of', 'view', 'text', 'b/']
['Participate', 'in', 'class,', "don't", 'sit', 'in', 'the', 'back,', 'and', "you'll", 'be', 'fine.']
['Once', 'i', 'couldnt', 'make', 'it', 'to', 'his', 'office', 'hours', 'and', 'he', 'told', 'me', 'to', 'prioritize', 'my', 'life!']
['You', 'will', 'need', 'to', 'go', 'to', 'the', 'study', 'groups', 'just', 'to', 'pass', 'let', 'alone', 'get', 'a', 'good', 'grade.']
['Oh', 'watch', 'out', 'for', 'his', 'hands', 'shake']
['The', 'questions', 'that', 'he', 'gives', 'to', 'discuss', 'at', 'study', 'group', 'he', 'bases', 'the', 'questions', 'for', 'his', 'tests', 'on.']
['Final', 'is', 'not', 'accumulative.']
['I', 'did', 'very', 'well', 'in', 'the', 'class,', 'but', 'I', 'know', 'people', 'who', 'needed', 'help', 'and', 'did', 'not', 'receive', 'it.']
['Only', 'take', 'this', 'class', 'if', 'you', 'have', 'an', 'American', 'Politics', 

['He', 'is', 'not', 'that', 'good.']
['Class', 'is', 'a', 'waste', 'of', 'time', 'and', 'practice', 'exams', 'are', 'apsolutely', 'nothing', 'like', 'his', 'real', 'exam.']
['Plus', 'hes', 'HOT']
['you', 'can', 'pass', 'the', 'class', 'without', 'going.']
['He', 'is', 'Hot!']
['Hot!']
['He', 'has', 'a', 'very', 'nice', 'accent.']
['He', 'is', 'helpful', 'and', 'cute', 'too.']
["It's", 'not', 'like', 'fall', 'or', 'spring.']
['You', 'have', 'to', 'memorize', '9', 'chapters', 'for', 'one', 'test', 'and', '4', 'for', 'the', 'final.']
['Easy', 'course,', 'Essay', 'questions', 'but', 'you', 'will', 'be', 'told', 'what', 'to', 'study', 'for.']
['Great', 'guy!!lectures', 'are', 'boring', 'and', 'tests', 'are', 'essays', 'but', 'he', 'will', 'point', 'out', 'what', 'you', 'need', 'to', 'study', 'for.']
['Lectures', 'are', 'boring.']
['Not', 'a', 'great', 'teacher,', 'but', 'is', 'there', 'a', 'good', 'teacher', 'for', 'this', 'class&#63;']
['Do', 'the', 'practice', 'quizzes', 'and', "you'll", 

['Michael', 'Burton', 'will', 'always', 'be', 'the', 'professor', 'I', 'look', 'back', 'upon', 'during', 'my', 'time', 'at', 'OU.']
['There', 'is', 'no', 'other', 'professor', 'in', 'the', 'poly', 'sci', 'dept', 'even', 'close', 'to', 'having', 'the', 'amount', 'of', 'political', 'professionalism', 'Burton', 'has.']
['OU', 'is', 'lucky', 'to', 'have', 'him.']
['I', "wouldn't", 'take', 'him', 'for', 'a', 'lower', 'level', 'course,', 'but', 'his', 'senior', 'level', 'courses', 'are', 'amazing.']
['Participate', 'in', 'class-']
['Take', 'Mike!']
['I', 'took', 'History', 'of', 'Politics', 'in', 'the', 'United', 'States', 'and', 'his', 'powerpoint', 'were', 'only', 'for', 'his', 'use,', 'so', 'it', 'was', 'strictly', 'oral', 'learning.']
['I', 'think', "it's", 'a', 'really', 'hard', 'class.']
['If', 'you', 'take', 'his', 'class', 'be', 'prepared', 'to', 'do', 'the', 'work.']
['Look,', 'Burton', 'is', 'smug', 'and', 'cocky,', 'but', 'he', 'knows', 'his', 'stuff.']
['You', 'will', 'have', 'su

In [50]:
dictionary = corpora.Dictionary(test_comments)

In [51]:
len(dictionary)

12146

In [52]:
print(dictionary.token2id)

{"It's": 0, 'class,': 1, 'for': 2, 'homeworks.': 3, 'in': 4, 'never': 5, 'only': 6, 'the': 7, 'used': 8, '10': 9, 'a': 10, 'every': 11, 'had': 12, 'minute': 13, 'other': 14, 'presentation': 15, 'we': 16, 'week.': 17, 'Very': 18, 'and': 19, 'helpful': 20, 'understanding': 21, 'able': 22, 'alwasy': 23, 'an': 24, 'example.': 25, "he's": 26, 'interesting': 27, 'is': 28, 'provide': 29, 'to': 30, 'Abnormal': 31, 'I': 32, 'have': 33, 'him': 34, 'now': 35, 'class': 36, 'fair': 37, 'interesting.': 38, 'keeps': 39, 'really': 40, 'think': 41, 'The': 42, 'from': 43, 'jumps': 44, 'lecture': 45, 'place': 46, 'place.': 47, 'Nice': 48, 'ambiguous': 49, 'are': 50, 'but': 51, 'guy': 52, 'tests': 53, 'very': 54, 'all': 55, 'all,': 56, 'distaster': 57, 'Had': 58, 'concerns.': 59, 'of': 60, 'psych': 61, 'ultimate': 62, 'VERY': 63, 'awesome.': 64, 'guess': 65, 'i': 66, 'ok': 67, 'was': 68, 'He': 69, 'Psychologists.': 70, 'as': 71, 'feel': 72, 'if': 73, 'makes': 74, 'real': 75, 'us': 76, 'AS': 77, 'BORING': 

In [53]:
model = Word2Vec(test_comments)

In [55]:
model.wv['lecture']

array([-0.00654152, -0.24468045, -0.22490175,  0.15102002, -0.401164  ,
       -0.29384831,  0.13530016, -0.02886784, -0.09186389,  0.16552174,
       -0.29071966, -0.14627807, -0.36789852,  0.45944312, -0.11952641,
       -0.19133486,  0.19616266,  0.10258605, -0.22294976, -0.31428587,
       -0.26958194,  0.10501771, -0.02173205,  0.00805537,  0.41023394,
        0.16130918, -0.32483569,  0.05007539,  0.11218302,  0.45972678,
        0.09441371, -0.33954859, -0.05941186, -0.03560461, -0.12046396,
       -0.1455593 ,  0.03150311,  0.03447314, -0.14023212,  0.19642596,
        0.06311397, -0.14176457, -0.04250157,  0.17564109, -0.26238576,
       -0.12930243,  0.1987074 ,  0.4033142 ,  0.04928567,  0.32300577,
       -0.24607494, -0.53290212, -0.4023917 ,  0.13959387,  0.26099855,
        0.22071479,  0.26815137,  0.30103514,  0.13786499,  0.28502226,
        0.09503521, -0.14093083, -0.01532888, -0.08879557,  0.02662169,
        0.363125  ,  0.05417698, -0.12060441,  0.11333493,  0.15

In [56]:
model.wv.most_similar('lecture')

[('information', 0.999783456325531),
 ('readings', 0.999717116355896),
 ('some', 0.9996970891952515),
 ('topics', 0.9996580481529236),
 ('during', 0.9996457099914551),
 ('everyone', 0.9996120929718018),
 ('But', 0.9996089935302734),
 ('material.', 0.9996076822280884),
 ('those', 0.9996045827865601),
 ('reading', 0.9996024370193481)]

In [58]:
model.wv.most_similar('teacher', topn =10)

[('professor', 0.9980145692825317),
 ("I've", 0.9896190166473389),
 ('my', 0.9885308742523193),
 ('one', 0.9876367449760437),
 ('teachers', 0.9875590801239014),
 ('worst', 0.9870314598083496),
 ('professors', 0.9868326187133789),
 ('was', 0.9852902889251709),
 ('had!', 0.9842613339424133),
 ('most', 0.9835221767425537)]

##### Testing Word2Vec model by removing stopwords and lemmetizing

In [63]:
f = open(test_filepath, 'r', encoding='utf-8')
test_comments = [f.readline()[:-1] for line in f]

In [65]:
for i, comment in enumerate(test_comments):
    lst = []
    processed_comment = nlp(comment)
    for word in processed_comment:
        if word.is_stop == False:
            lst.append(word.lemma_)
    test_comments[i] = lst

In [66]:
test_comments[6]

['the', 'lecture', 'jump', 'from', 'place', 'to', 'place', '.']

In [67]:
for sent in test_comments:
    print(sent)

['-PRON-', 'be', 'never', 'use', 'in', 'class', ',', 'only', 'for', 'the', 'homework', '.']
['-PRON-', 'have', 'a', '10', 'minute', 'presentation', 'every', 'other', 'week', '.']
['very', 'helpful', 'and', 'understanding']
['-PRON-', 'be', 'interesting', 'and', 'alwasy', 'be', 'able', 'to', 'provide', 'an', 'example', '.']
['-PRON-', 'have', '-PRON-', 'now', 'for', 'abnormal']
['-PRON-', 'think', '-PRON-', 'be', 'really', 'fair', 'and', 'keep', 'the', 'class', 'interesting', '.']
['the', 'lecture', 'jump', 'from', 'place', 'to', 'place', '.']
['nice', 'guy', 'but', 'the', 'test', 'be', 'very', 'ambiguous']
['all', 'and', 'all', ',', 'a', 'distaster']
['have', '-PRON-', 'for', 'psych', 'of', 'ultimate', 'concern', '.']
['very', 'interesting', '.']
['-PRON-', 'be', 'awesome', '.']
['be', 'ok', 'i', 'guess']
['-PRON-', 'make', '-PRON-', 'feel', 'as', 'if', '-PRON-', 'be', 'real', 'psychologist', '.']
['boring', 'as', 'hell', '!']
['run', 'in', 'the', 'opposite', 'direction', '!']
['study'

['-PRON-', 'be', 'also', 'very', 'good', 'at', 'the', '"', 'society', '"', 'part', 'and', 'try', 'hard', 'to', 'make', 'everything', 'relevent', '.']
['-PRON-', 'teach', 'like', '-PRON-', 'be', 'high', 'school', 'student', '-', 'assign', 'homework', 'daily', ',', 'call', 'on', 'almost', 'everyone', 'during', 'one', 'class', 'period', ',', 'and', 'give', 'a', 'lab', 'report', 'every', 'week', '.']
['the', 'material', 'be', 'super', 'easy', ';', '-PRON-', 'be', 'just', 'a', 'lot', 'of', 'tedious', 'work', '.']
['the', 'class', 'can', 'be', 'funny', 'if', '-PRON-', 'have', 'a', 'good', 'sense', 'of', 'humor', 'about', '-PRON-', '-', '-PRON-', 'can', 'get', 'ridiculous', '.']
['dr.', 'berkow', 'post', 'all', 'power', '-', 'point', 'slide', 'on', 'blackboard', '..']
['exam', 'be', 'a', 'bit', 'random', '--', 'and', 'no', 'matter', 'what', '-PRON-', 'say', '--', '-PRON-', 'study', 'guide', 'be', 'not', 'helpful', '.']
['read', 'the', 'book', ',', '-PRON-', 'will', 'be', 'fine', '.']
['will',

['and', '-PRON-', 'know', 'what', '-PRON-', 'be', 'talk', 'about', '.']
['no', 'book', 'require', ',', 'just', 'a', 'propacket', 'of', '-PRON-', 'slide', 'show', ',', 'so', '-PRON-', 'barely', 'have', 'to', 'take', 'note', '!']
['dr.', 'shari', 'robertson', 'be', '"', 'everybody', "'s", 'mom', '"', '.']
['-PRON-', 'really', 'tire', 'to', 'watch', 'out', 'for', 'everyone', '.']
['there', 'be', 'two', 'mid', '-', 'semester', 'exam', ',', 'a', 'final', ',', 'a', 'lesson', 'plan', 'project', ',', 'in', 'class', 'activity', ',', 'and', 'attendence', 'count', '.']
['not', 'really', 'hard', ',', 'just', 'really', 'boring', '.']
['-PRON-', 'reccommend', 'if', '-PRON-', 'want', 'an', 'a', ',', 'but', 'just', 'be', 'prepared', 'to', 'want', 'to', 'fall', 'asleep', 'every', 'day']
['-PRON-', 'be', 'awesome', '!', '!']
['this', 'woman', 'amaze', '-PRON-', '.']
['-PRON-', 'be', 'very', 'knowledgeable', 'in', '-PRON-', 'field', 'and', '-PRON-', 'convey', 'in', 'the', 'information', 'in', 'such', 'a'

['-PRON-', 'be', 'a', 'bit', 'of', 'a', 'dork', ',', '-PRON-', 'can', 'definitly', 'tell', '-PRON-', 'have', 'see', 'lord', 'of', 'the', 'rings', 'one', 'too', 'many', 'time', 'and', 'still', 'play', 'dungeons', 'and', 'dragons', 'with', '-PRON-', 'buddies[which', '-PRON-', 'have', 'confess', 'to', 'by', 'the', 'way', ']']
['funny', 'and', 'odd', 'guy', '.']
['do', 'not', 'read', 'so']
['-PRON-', 'voice', 'boom', 'off', 'the', 'back', 'wall']
['-PRON-', 'be', 'take', 'another', 'class', 'with', '-PRON-', 'next', 'semester']
['dr', 'rice', 'be', 'a', 'little', 'obsessive', 'about', 'some', 'thing', ',', '-PRON-', 'be', 'overal', 'helpful', 'but', '-PRON-', 'can', 'not', 'slack', 'at', 'all']
['-PRON-', 'be', 'a', 'good', 'prof', '.']
['-PRON-', 'be', 'cool', ',', '-PRON-', 'help', '-PRON-', 'and', 'stuff']
['but', 'prepare', '-PRON-', 'keep', '-PRON-', 'for', 'the', 'whole', 'class', 'time', '.']
['which', 'seem', 'awsome', ',', 'but', 'when', '-PRON-', 'six', 'at', 'night', 'and', '-PR

In [68]:
dictionary = corpora.Dictionary(test_comments)

In [69]:
len(dictionary)

5528

In [70]:
print(dictionary.token2id)

{',': 0, '-PRON-': 1, '.': 2, 'be': 3, 'class': 4, 'for': 5, 'homework': 6, 'in': 7, 'never': 8, 'only': 9, 'the': 10, 'use': 11, '10': 12, 'a': 13, 'every': 14, 'have': 15, 'minute': 16, 'other': 17, 'presentation': 18, 'week': 19, 'and': 20, 'helpful': 21, 'understanding': 22, 'very': 23, 'able': 24, 'alwasy': 25, 'an': 26, 'example': 27, 'interesting': 28, 'provide': 29, 'to': 30, 'abnormal': 31, 'now': 32, 'fair': 33, 'keep': 34, 'really': 35, 'think': 36, 'from': 37, 'jump': 38, 'lecture': 39, 'place': 40, 'ambiguous': 41, 'but': 42, 'guy': 43, 'nice': 44, 'test': 45, 'all': 46, 'distaster': 47, 'concern': 48, 'of': 49, 'psych': 50, 'ultimate': 51, 'awesome': 52, 'guess': 53, 'i': 54, 'ok': 55, 'as': 56, 'feel': 57, 'if': 58, 'make': 59, 'psychologist': 60, 'real': 61, '!': 62, 'boring': 63, 'hell': 64, 'direction': 65, 'opposite': 66, 'run': 67, 'about': 68, 'culture': 69, 'do': 70, 'eastern': 71, 'kow': 72, 'much': 73, 'not': 74, 'so': 75, 'study': 76, 'bot': 77, 'easy': 78, 'fu

In [71]:
model = Word2Vec(test_comments)

In [72]:
model.wv['lecture']

array([ -8.14721435e-02,  -2.54949450e-01,   1.22446530e-01,
         3.06313068e-01,  -1.04968816e-01,  -1.72160193e-01,
        -1.29253268e-01,   2.27997694e-02,  -3.33239645e-01,
        -1.69440191e-02,  -3.31437230e-01,  -2.17524976e-01,
        -2.92112827e-01,   4.55457449e-01,  -3.68775070e-01,
         1.04791112e-01,   1.13101035e-01,   4.96119633e-02,
        -1.42382190e-01,  -5.23562551e-01,  -7.55129084e-02,
         3.52280475e-02,   1.81204081e-02,   8.62910375e-02,
         2.54569560e-01,   1.16353177e-01,  -2.20348269e-01,
        -8.42319727e-02,   4.86138687e-02,   2.97865331e-01,
         2.43759096e-01,  -1.96211532e-01,   9.29939747e-02,
        -3.43820840e-01,  -3.00359502e-02,  -4.66552049e-01,
        -2.05552921e-01,   1.46650568e-01,  -2.95070469e-01,
         2.28216693e-01,   2.81434953e-01,  -2.74980575e-01,
         3.25369805e-01,   9.46044922e-02,  -4.42791253e-01,
        -2.18400359e-01,  -6.16051198e-04,   2.17768729e-01,
        -3.97884309e-01,

In [73]:
model.wv.most_similar('lecture')

[('explain', 0.9856079816818237),
 ('reading', 0.9854652285575867),
 ('everything', 0.9841817021369934),
 ('material', 0.9841325283050537),
 ('and', 0.9792267084121704),
 (',', 0.9786258935928345),
 ('always', 0.9785717725753784),
 ('thing', 0.9768021106719971),
 ('prepare', 0.9765354990959167),
 ('follow', 0.976283609867096)]

In [74]:
model.wv.most_similar('teacher', topn =10)

[('professor', 0.98128741979599),
 ('guy', 0.9795665144920349),
 ('prof', 0.9755983352661133),
 ('great', 0.9639059901237488),
 ('awesome', 0.9559733867645264),
 ('zeakes', 0.9464095830917358),
 ('amazing', 0.9381518959999084),
 ('dr.', 0.9343852996826172),
 ('man', 0.9312210083007812),
 ('good', 0.9244221448898315)]

In [75]:
model.wv.most_similar('learn', topn =10)

[('talk', 0.9811581373214722),
 ('want', 0.9791783094406128),
 ('know', 0.9722205400466919),
 ('care', 0.9719315767288208),
 ('understand', 0.9698665142059326),
 ('much', 0.9657596349716187),
 ('expect', 0.9637631773948669),
 ('well', 0.9594025611877441),
 ('teach', 0.9591293334960938),
 ('student', 0.9590266346931458)]

In [76]:
model.wv.most_similar('teach', topn =10)

[('think', 0.9793863892555237),
 ('stuff', 0.9790608286857605),
 ('tell', 0.9782949686050415),
 ('can', 0.9781153202056885),
 ('how', 0.97685706615448),
 ('like', 0.975929856300354),
 ('try', 0.9757285118103027),
 ('expect', 0.9726259708404541),
 ('anything', 0.971917986869812),
 ('say', 0.9684455990791321)]

In [78]:
dictionary.token2id['teach']

163

In [79]:
model.wv.most_similar('professor', topn =10)

[('teacher', 0.98128741979599),
 ('prof', 0.9769079685211182),
 ('awesome', 0.9538158774375916),
 ('ever', 0.9524438977241516),
 ('zeakes', 0.9474201202392578),
 ('amazing', 0.9457699060440063),
 ('dr.', 0.944303035736084),
 ('great', 0.9357653260231018),
 ('bad', 0.9318722486495972),
 ('guy', 0.9309294819831848)]